src/ClientData/include/ClientData/CallstackData.h

```diff
@@ -83,9 +83,32 @@ class CallstackData {
   template <typename Action>
   void ForEachCallstackEventInTimeRangeDiscretized(uint64_t min_timestamp, uint64_t max_timestamp,
                                                    uint32_t resolution, Action&& action) const {
-    for (const auto& [tid, unused_events] : callstack_events_by_tid_) {
-      ForEachCallstackEventOfTidInTimeRangeDiscretized(tid, min_timestamp, max_timestamp,
-                                                       resolution, action);
+    auto get_next_callstack = [&](uint64_t timestamp) -> CallstackEvent {
+      constexpr uint32_t kFakeId = -1;
+      CallstackEvent next_callstack{max_timestamp, kFakeId, kFakeId};
```

Comment:
Why can't we just return `optional<CallstackEvent>` instead of instroducing yet another special constant?

Reply:
This should be be either `std::optional` (if owning) or a possibly-null pointer (if non-owning). See http://go/totw/171

Reply:
This constants are not used, it's just to fill the callstack with some ID. Initially I implemented as an optional but the condition was a bit harder to read. But I agree that using max_timestamp here is strange, will convert it back to optional.